rules.go

```diff
@@ -0,0 +1,250 @@
+package main
+
+import (
+	"fmt"
+	"slices"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var timeFormat = "20060102"
+
+// daysInMonth формирует мапу допустимых значений дней в месяце
+func daysInMonth(year int) map[int]int {
+	daysAmout := make(map[int]int)
+	for month := 1; month <= 12; month++ {
+		nextMonth := time.Date(year, time.Month(month)+1, 1, 0, 0, 0, 0, time.Local)
+		days := nextMonth.AddDate(0, 0, -1).Day()
+		daysAmout[month] = days
+	}
+	return daysAmout
+}
+
+// DateParse возвращает ближайшее время задачи
+func DateParse(now time.Time, dateStr string, repeat string) (string, error) {
+	date, err := time.Parse(timeFormat, dateStr)
+	if err != nil {
+		return "", fmt.Errorf("ошибка при парсинге времени date: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
		return "", fmt.Errorf("ошибка при парсинге времени date: %w", err)
```